### DISCUSSION TITLE
react-navigation 4 to  6 migration

### DISCUSSION THREAD

**Discussion by gkasireddy202** [react-navigation/react-navigation#11960] [General]
Hi,

We are using the react-navigation 4 in my project. All screens are class components in my project.
How do you migrate from react-navigation from 4 to 6?

**Comment by raajnadar** [comment]
Docs to migrate from v4 to v5 => https://reactnavigation.org/docs/5.x/upgrading-from-4.x

Docs to migrate from v5 to v6 => https://reactnavigation.org/docs/upgrading-from-5.x

You can read the above docs to migrate, if you want to preserve the old API and still use the v5 version you can try this https://reactnavigation.org/docs/5.x/compatibility

**Reply by gkasireddy202** [reply]
@raajnadar - Can I directly migrate from navigation 4 to 6?

**Reply by raajnadar** [reply]
There is no compact layer in v6, if you can migrate to v5 going to v6 is very simple. But you cannot go to v6 without migrating to the new API because there is no compact layer package in v6.

**Reply by gkasireddy202** [reply]
@raajnadar  - Thanks for your reply.
if we migrate to v5 and later v6, it will take more time for development and testing?
Need your thoughts?
Can we migrate to v6 without using a compact layer?

**Reply by raajnadar** [reply]
All the answers depend on your codebase, if you don't want to move to v5, you have the code of the v4 fork to solve the issue and use it in your project from your fork.

I always recommend upgrading to the latest version so you don't miss important updates. So now it is up to you to migrate or stay with the lower version of react native, react native 0.70 should work for v4.

**Reply by gkasireddy202** [reply]
@raajnadar - Thanks.

Can we migrate to v6 or v7?

**Reply by raajnadar** [reply]
Do you read comments before replying?

**Reply by gkasireddy202** [reply]
Yes.Need suggestions for migration.
We already upgraded my project to react-native 0.73.2.

Which navigation version is better to migrate v6 or v7?

**Reply by raajnadar** [reply]
Migrate to v5 first then only it will be easier to migrate to v6

**Reply by gkasireddy202** [reply]
@raajnadar - Thanks
